src/server/races/default-race.js

```diff
@@ -0,0 +1,279 @@
+			if(	config.discord.enabled
+				&& skins.skinList[skinId].premium
+				&& !permissions.memberHasPermission(userId, "PREMIUM_SKINS")) {
+				return undefined;
+			}
+			return skinId;
+		};
+		this.defaultMarbleAttributeList.skinId = function() {
+			return "default";
+		};
+
+		// 2% chance for a entry's marbles to be of a random size between .3 and .6
+		this.defaultMarbleAttributeList.size = function() {
+			return Math.random() > .98 ? (.3 + Math.random() * .3) : 0.2;
+		};
+	}
+
+	//
+	// State-related functions
+	//
+
+	onStateEnter() {
+	}
+
+	// Returns a promise that resolves once preparation has finished, or null if this state can be skipped
+	onStatePreparing() {
+		return null;
+	}
+
+	onStateStarting() {
+		// Default point contribution rules
+		// Other race types may want to override this with their own
+		this.pointContributors = this.raceEntries.filter(entry => entry.contributesPoints).length;
+	}
+
+	onStateStarted() {
+		this.round.start = Date.now();
+		physicsWorld.openGates();
+	}
+
+	onStateFinished() {
+		this.finalizeRound();
+
+		physicsWorld.clearMarbles();
+		physicsWorld.closeGates();
+	}
+
+	//
+	// Non-state functions
+	//
+
+	// Returns true if the marble/entry limit has been reached
+	addRaceEntry(id, name, attributeString = "") {
+		// Don't add new player entries if the limit has been reached, or players that already entered
+		// Exception: Added bots with a undefined id (only spawnable by devs/operators)
+		if(typeof id === "string" && this.isRaceFull()
+		|| (typeof id === "string" && typeof this.raceEntries.find(playerEntry => id === playerEntry.id) !== "undefined") )
+			return this.isRaceFull();
+
+		let marbleAttributes = this.parseMarbleAttributes(id, attributeString);
+		let entry = new RaceEntry(id, name, marbleAttributes);
+		this.raceEntries.push(entry);
+
+		this.onNewRaceEntry(entry);
+
+		return this.isRaceFull();
+	}
+
+	// Returns a marbleAttributes object for the given user id and attributes string
+	parseMarbleAttributes(id, attributeString) {
+		let marbleAttributes = {};
+		let messageSections = attributeString.split(" ");
+		messageSections.length = Math.min(messageSections.length, 4); // At most, check only the first 4 words
+
+		for(let i = 0; i < messageSections.length; i++) {
+			Object.keys(this.marbleAttributeList).forEach( attribute => {
+				// If this attribute hasn't been set yet, check for a match and set the property if we have a valid return value
+				if(typeof marbleAttributes[attribute] === "undefined") {
+					let match = this.marbleAttributeList[attribute](messageSections[i], id);
+					if(typeof match !== "undefined") {
+						marbleAttributes[attribute] = match;
+					}
+				}
+			});
+		}
+
+		// Set any default attributes that are (still) undefined
+		Object.keys(this.defaultMarbleAttributeList).forEach( attribute => {
+			if(typeof marbleAttributes[attribute] === "undefined") {
+				marbleAttributes[attribute] = this.defaultMarbleAttributeList[attribute]();
```

Comment:
There already is a reason for this to be called as functions: the default randomized `color` and `size` which are defined above. Other game modes could also have reason to use this (e.g. for auto-joining teams, themed skins, whatever else it could come up with), so this adds flexibility for those cases.
I can understand the confusion with `default` part of the name though, so let me know if you have suggestions for alternatives.

Reply:
Ah yes, oversight on my part. I expected the default to just be a static value. Not sure on any alternative names though 🤔